pkg/hash_cache.go

```diff
@@ -519,20 +527,20 @@ func hashRule(thc *TargetHashCache, rule *build.Rule, configuration *analysis.Co
 					return nil, fmt.Errorf("configuredRuleInputs for %s included %s in configuration %s but it couldn't be found either unconfigured or in the depending target's configuration %s. This probably indicates a bug in Bazel - please report it with a git repo that reproduces at https://github.com/bazel-contrib/target-determinator/issues so we can investigate", rule.GetName(), ruleInputLabel, ruleInputConfiguration, ownConfiguration)
 				}
 			}
-			ruleInputHash, err := thc.Hash(LabelAndConfiguration{Label: ruleInputLabel, Configuration: ruleInputConfiguration})
-			if err != nil {
-				return nil, fmt.Errorf("failed to hash configuredRuleInput %s %s which is a dependency of %s %s: %w", ruleInputLabel, ruleInputConfiguration, rule.GetName(), configuration.GetChecksum(), err)
-			}
-			writeLabel(hasher, ruleInputLabel)
-			hasher.Write(ruleInputConfiguration.ForHashing())
-			hasher.Write(ruleInputHash)
+			labelsAndConfigurations = append(labelsAndConfigurations, LabelAndConfigurations{
+				Label:          ruleInputLabel,
+				Configurations: []Configuration{ruleInputConfiguration},
```

Comment:
Just a passing thought: would it make sense to use a pointer for the configurations? Seems like a config hash string is 64 bytes currently (vs typically 8 bytes for a pointer), and since it's repeated all over the place it could save a fair amount of memory. Not sure what would be the impact on adding a de-reference each time we read the value though.

Reply:
This has actually been on my to do list! Bazel has deprecated using configuration hashes and switching to integer IDs in a list of configurations (https://github.com/bazelbuild/bazel/blob/47e4493f62a457c95f5828dcce748377a08d894a/src/main/protobuf/analysis_v2.proto#L219-L225)

The new index field was added in 6.0.0 (https://github.com/bazelbuild/bazel/commit/29d46eb41a413430498ab033f84a8d960edfb6fb) so as soon as we're happy to stop supporting 5.x we can just switch the code over to use IDs completely.

I figured it wasn't worth effectively implementing our own interning when we can hopefully achieve the same effect at some point via Bazel-supplied interning, but we could do this ourselves if we wanted to be more eager.

Reply:
Makes a lot of sense! I'm sure we can wait until Bazel 5.x get deprecated 👍